features/mbedtls/platform/inc/mbedtls_config_reader.h

```diff
@@ -0,0 +1,500 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __MBEDTLS_CONFIG_READER_H__
+#define __MBEDTLS_CONFIG_READER_H__
+
+#include "platform/inc/platform_mbed.h"
+
+#if !defined(MBEDTLS_ENTROPY_HARDWARE_ALT) && !defined(MBEDTLS_TEST_NULL_ENTROPY) && \
+    !defined(MBEDTLS_ENTROPY_NV_SEED)
+    // for platforms that don't have entropy,none of the ciphersuites will
+    // work, so don't bother
+#undef MBED_CONF_MBEDTLS_PELION_CLIENT
+#undef MBED_CONF_MBEDTLS_MBED_PELION_MINI_CLIENT
+#undef MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+#undef MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
+#undef MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384
+#undef MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+#undef MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_CCM
+#undef MBED_CONF_MBEDTLS_PSK_WITH_AES_CCM
+#undef MBED_CONF_MBEDTLS_PSK_WITH_AES_128_CBC_SHA256
+#undef MBED_CONF_MBEDTLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256
+#undef MBED_CONF_MBEDTLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256
+#undef MBED_CONF_MBEDTLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384
+#undef MBED_CONF_MBEDTLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384
+#undef MBED_CONF_MBEDTLS_PSK_WITH_AES_128_GCM_SHA256
+#undef MBED_CONF_MBEDTLS_PSK_WITH_AES_256_GCM_SHA384
+#undef MBED_CONF_MBEDTLS_PSK_WITH_AES_256_CBC_SHA384
+#undef MBED_CONF_MBEDTLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
+#undef MBED_CONF_MBEDTLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+#undef MBED_CONF_MBEDTLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
+#undef MBED_CONF_MBEDTLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
+#undef MBED_CONF_MBEDTLS_RSA_WITH_AES_256_GCM_SHA384
+#undef MBED_CONF_MBEDTLS_RSA_WITH_AES_128_GCM_SHA256
+#undef MBED_CONF_MBEDTLS_RSA_WITH_AES_128_CBC_SHA256
+#undef MBED_CONF_MBEDTLS_RSA_WITH_AES_256_CBC_SHA256
+#undef MBED_CONF_MBEDTLS_ECDH_RSA_WITH_AES_128_CBC_SHA256
+#undef MBED_CONF_MBEDTLS_ECDH_RSA_WITH_AES_128_GCM_SHA256
+#undef MBED_CONF_MBEDTLS_ECDH_RSA_WITH_AES_256_CBC_SHA384
+#undef MBED_CONF_MBEDTLS_ECDH_RSA_WITH_AES_256_GCM_SHA384
+    
+#endif // !defined(MBEDTLS_ENTROPY_HARDWARE_ALT) && !defined(MBEDTLS_TEST_NULL_ENTROPY) && !defined(MBEDTLS_ENTROPY_NV_SEED)
+
+// This first section is checking the values from mbed_app.json and
+// mbedtls/mbed_lib.json and setting the appropriate defines for them
+
+#if MBED_CONF_MBEDTLS_PELION_CLIENT
+    #undef MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+    #define MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 1
+    #undef MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+    #define MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384 1
+#endif
+
+#if MBED_CONF_MBEDTLS_MBED_PELION_MINI_CLIENT
+    #undef MBED_CONF_MBEDTLS_PSK_WITH_AES_CCM
+    #define MBED_CONF_MBEDTLS_PSK_WITH_AES_CCM 1
+    #undef MBED_CONF_MBEDTLS_PSK_WITH_AES_128_CBC_SHA256
+    #define MBED_CONF_MBEDTLS_PSK_WITH_AES_128_CBC_SHA256 1
+#endif
+#if MBED_CONF_MBEDTLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
```

Comment:
Style: Please be consistent. If there is a new line in between everyone of this blocks, please make sure there is always a newline.

Reply:
will fix